### PR TITLE
Issue https://github.com/ATFutures/calendar/issues/57 fix :wrench: 

### PR DESCRIPTION
**Pull request by serkor1**
Hi,

I have created a new `R`-script called `utils`, which contains the `assert()`-function. For some devs this is the standard file to put these kind of helper-functions in - this is at least what I do :smile: 

> [!NOTE]
> I have only added the functionality to functions that already used a mix of `stopifnot` and `stop`. I haven't added this functionality on any other functions.

I found two typos that I have corrected also. See commit e63371e8787082c58ed5227b3236b2b5bc2b4f6c and 24c12111991becc56b4da14377f24188e440962e. 

The dependency on {methods} are gone, and a new dependency on {cli} is made. The {methods}-package were only used in two functions. So I believe this was a good choice.

I have run `devtools::check()` and everything passes locally.

Best,

@serkor1

### CODE REVIEW

**Comment by Robinlovelace** [comment]
Great stuff, will take a look!

File: DESCRIPTION [hunk 1]
```diff
@@ -1,5 +1,5 @@
 Package: calendar
-Title: Create, Read, Write, and Work with 'iCalander' Files, Calendars and
+Title: Create, Read, Write, and Work with 'iCalendar' Files, Calendars and
```
Comment (Robinlovelace) [inline DESCRIPTION:2]: Woa typo in the title? Not sure how that slipped the net..

File: DESCRIPTION [hunk 2]
```diff
@@ -22,9 +22,15 @@ Authors@R:
              family = "Scarafia",
              role = "ctb",
              email = "franco.scarafia@hotmail.com",
-             comment = c(ORCID = "0009-0005-9822-169X")))
+             comment = c(ORCID = "0009-0005-9822-169X")),
+      person(given   = "Serkan",
```
Comment (Robinlovelace) [inline DESCRIPTION:26]: :+1:

File: R/ic_dataframe.R [hunk 3]
```diff
@@ -17,15 +17,24 @@
...
-  if (methods::is(object = x, class2 = "data.frame")) {
+  if (inherits(x = x, what = "data.frame")) {
 
     return(x)
 
   }
 
-  stopifnot(methods::is(object = x, class2 = "character") | methods::is(object = x, class2 = "list"))
+  assert(
+    inherits(x = x, what = "character") | inherits(x = x, what = "list"),
```
Comment (Robinlovelace) [inline R/ic_dataframe.R:27]: :+1:

File: R/ical.R [hunk 4]
```diff
@@ -54,20 +85,23 @@ ical <- function(x, ic_attributes = NULL) {
 ic_attributes_vec <- function(
   x = NULL,
   ic_attributes = c(
-    BEGIN = "VCALENDAR",
-    PRODID = "ATFutures/calendar",
-    VERSION = "2.0",
+    BEGIN    = "VCALENDAR",
```
Comment (Robinlovelace) [inline R/ical.R:88]: Why this extra whitespace?

**Review by Robinlovelace — APPROVED** [review]
Looks good to me. Any extra comments @layik, @mpadge or anyone else before merging? See comments from me @serkor1. Good work :rocket:.

File: R/ical.R (hunk 4 above)
Comment (serkor1) [reply R/ical.R:88]: I don't know if it's redundant. But I thought it "looked better" honestly. I changed it from,

```
a = 2
aa = 2
aaa = 2
```

to 

```
a   = 2
aa  = 2
aaa = 2
```

File: DESCRIPTION (hunk 1 above)
Comment (serkor1) [reply DESCRIPTION:2]: It happens - I misspelled my own name once! 🤷‍♀️

**Comment by layik** [comment]
> Looks good to me. Any extra comments @layik, @mpadge or anyone else before merging? See comments from me @serkor1. Good work 🚀.

I am away so quick reply from me to say if it passes the tests (coverage of which I just checked is now deactivated on CodeCov) it is good enough for me. Thanks @serkor1 for the contribution.

**Pull request merged**